engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/query/NadelFieldToService.kt

```diff
@@ -28,6 +29,21 @@ internal class NadelFieldToService(
         }
     }
 
+    /**
+     * Returns the dynamically resolved service for the field, if it is annotated with @dynamicServiceResolution,
+     * otherwise returns the originalService.
+     */
+    fun resolveDynamicService(
```

Comment:
Can't we just integrate this into `getServicesForTopLevelFields` as an abstracted detail or do you think that's a bad idea?

e.g.

```kotlin
    fun getServicesForTopLevelFields(query: ExecutableNormalizedOperation): List<NadelFieldAndService> {
        return query.topLevelFields.flatMap { topLevelField ->
            if (dynamicServiceResolution.needsDynamicServiceResolution(field)) { // NEW
                dynamicServiceResolution.resolveServiceForField(field) // NEW
            } else if (isNamespacedField(topLevelField)) {
                getServicePairsForNamespacedFields(topLevelField)
            } else {
                listOf(getServicePairFor(field = topLevelField))
            }
        }
    }
```

That way, we don't need to add code into `NextgenEngine` specifically to deal with that.

Reply:
That would simplify the code yeah. 
The only problem is that if something goes wrong in the process of dynamically resolving the service for a field, the whole execution would blow up - which is what happens currently when a regular service association fails in `getServicesForTopLevelFields`.
With the current state of the code, only the field that uses dynamic service resolution would fail, so the execution could still be a partial success.

Reply:
Ahh I see, sorry didn't see this comment yesterday. That makes sense. Kinda annoying but yep, sounds good.

Let's just merge this as is then.